ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py

```diff
@@ -96,6 +96,9 @@ def __init__(
         )
 
         self._table = self._convert_table_to_orm_object(sqa_metadata)
+        self.set_session()
+
+    def set_session(self):
         self.session_factory = self._session_factory()
         self.session = self.session_factory()
         self.set_session_tag(self.session)
```

Comment:
unsure if we should do that. I know we do it in quite some places but it is generally not a good practice to instantiate object attributes outside of the `__init__`

Reply:
the method instantiates the object attributes which is called from __init__ itself and instantiates before being used, and also can be used again to instantiate object attributes with new connection in different methods..

Reply:
reverted this as it's not needed, thanks